pkg/crc/config/viper_config.go

```diff
@@ -98,33 +116,54 @@ func InitViper() error {
 	v.SetTypeByDefaultValue(true)
 	err := v.ReadInConfig()
 	if err != nil {
-		return fmt.Errorf("Error reading configuration file '%s': %v", constants.ConfigFile, err)
+		return nil, fmt.Errorf("Error reading configuration file '%s': %v", constants.ConfigFile, err)
 	}
 	v.WatchConfig()
-	globalViper = v
-	return v.Unmarshal(&changedConfigs)
+	var changedConfigs map[string]interface{}
+	if err := v.Unmarshal(&changedConfigs); err != nil {
+		return nil, err
+	}
+	cfg := &ViperConfig{
+		globalViper:    v,
+		changedConfigs: changedConfigs,
+		allSettings:    make(map[string]*Setting),
+	}
+
+	// Start command settings in config
+	cfg.AddSetting(Bundle, nil, []ValidationFnType{ValidateBundle}, []SetFn{SuccessfullyApplied})
+	cfg.AddSetting(CPUs, constants.DefaultCPUs, []ValidationFnType{ValidateCPUs}, []SetFn{RequiresRestartMsg})
+	cfg.AddSetting(Memory, constants.DefaultMemory, []ValidationFnType{ValidateMemory}, []SetFn{RequiresRestartMsg})
+	cfg.AddSetting(NameServer, nil, []ValidationFnType{ValidateIPAddress}, []SetFn{SuccessfullyApplied})
+	cfg.AddSetting(PullSecretFile, nil, []ValidationFnType{ValidatePath}, []SetFn{SuccessfullyApplied})
+	cfg.AddSetting(DisableUpdateCheck, nil, []ValidationFnType{ValidateBool}, []SetFn{SuccessfullyApplied})
+	// Proxy Configuration
+	cfg.AddSetting(ExperimentalFeatures, nil, []ValidationFnType{ValidateBool}, []SetFn{SuccessfullyApplied})
+	cfg.AddSetting(HTTPProxy, nil, []ValidationFnType{ValidateURI}, []SetFn{SuccessfullyApplied})
+	cfg.AddSetting(HTTPSProxy, nil, []ValidationFnType{ValidateURI}, []SetFn{SuccessfullyApplied})
+	cfg.AddSetting(NoProxy, nil, []ValidationFnType{ValidateNoProxy}, []SetFn{SuccessfullyApplied})
```

Comment:
Can this change be split in a separate commit, so that the introduction of the `ViperConfig` type is a fairly mechanical change which can be reviewed without thinking too much?